lib/ecto/migrator.ex

```diff
@@ -177,14 +214,36 @@ defmodule Ecto.Migrator do
     versions = migrated_versions(repo)
 
     Enum.map(pending_in_direction(versions, directory, :down) |> Enum.reverse, fn {a, b, _}
-     -> {:up, a, b}
+      -> {:up, a, b}
     end)
     ++
     Enum.map(pending_in_direction(versions, directory, :up), fn {a, b, _} ->
       {:down, a, b}
     end)
   end
 
+  defp lock_schema_migrations(repo, opts, fun) do
+    repo.transaction(fn ->
+      # TODO: Move to adapter
+      Ecto.Query.from(p in {SchemaMigration.get_source(repo), SchemaMigration}, select: p.version, lock: "FOR UPDATE")
+      |> Map.put(:prefix, opts[:prefix])
+      |> repo.all(timeout: :infinity, log: false)
+
+      try do
```

Comment:
It seems this try catch is also not necessary.

Reply:
The reason for this try catch is so that if an exception occurs on the 5th migration, we still commit the schema migrations transaction, which records that we did the first 4.